gitpod-io/gitpod
# independent deployment of code images

## Description


This PR extracts an ide configmap for the server to allow deploy it independent from other k8s resources by Argo CD.

## Related Issue(s)

Fixes #5565

## How to test


- Test that you can start a workspace: https://ak-code-deployment.staging.gitpod-dev.com/#github.com/akosyakov/parcel-demo
- Apply a bogus configuration:
```
kubectl get cm server-ide-config -o yaml | \
  sed -e 's|ideVersion|ideVersion2|' | \
  kubectl apply -f -
```
- You may need to wait a bit, but eventually the server should log errors like: `Error: invalid: data should have required property 'ideVersion'`.
- Despite it though you should be able to start new workspaces.
- Apply valid configuration:
```
kubectl get cm server-ide-config -o yaml | \
  sed -e 's|ideVersion2|ideVersion|' | \
  sed -e 's|commit-2f069ca3b646ab8099c05f7efd181004c08e95cb|commit-01948796e3c2d0a42ba9e63abe483b62941de53d|' | \
  kubectl apply -f -
```
- You may need to wait a bit, but eventually the server should log: `ide config updated`.
- Start a new workspace and check that VS Code now is of version 1.60 in the about dialog.

## Release Notes

```release-note
NONE
```
